test/e2e/tests/test_retain_policy.py

```diff
@@ -0,0 +1,113 @@
+
+
+@pytest.fixture(scope="module")
+def retained_notebook_instance_lifecycle_config():
+    notebook_instance_lfc_name = random_suffix_name("notebookinstancelfc", 40)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["NOTEBOOK_INSTANCE_LFC_NAME"] = notebook_instance_lfc_name
+    replacements["DELETION_POLICY"] = "retain"
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural=RESOURCE_PLURAL,
+        resource_name=notebook_instance_lfc_name,
+        spec_file=RESOURCE_SPEC_FILE,
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource, spec)
+
+    deletion_resp = delete_notebook_instance_lifecycle_config(
+        notebook_instance_lfc_name
+    )
+    assert deletion_resp is not None
+
+
+def get_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().describe_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+def delete_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().delete_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+@service_marker
+class TestRetainPolicy:
+    def test_retain_notebook_instance_lifecycle(
+        self, retained_notebook_instance_lifecycle_config
+    ):
+        (reference, resource, _) = retained_notebook_instance_lifecycle_config
+        assert k8s.get_resource_exists(reference)
+
+        # Getting the resource name
+        notebook_instance_lfc_name = resource["spec"].get(
+            "notebookInstanceLifecycleConfigName", None
+        )
+        assert notebook_instance_lfc_name is not None
+        notebook_instance_lfc_desc = get_notebook_instance_lifecycle_config(
+            notebook_instance_lfc_name
+        )
+        assert (
+            k8s.get_resource_arn(resource)
+            == notebook_instance_lfc_desc["NotebookInstanceLifecycleConfigArn"]
+        )
+
+        # Delete the CR in Kubernetes
+        _, deleted = k8s.delete_custom_resource(reference)
+
+        assert deleted
+
+        # Verify that the resource was not deleted on SageMaker
+        assert (
+            get_notebook_instance_lifecycle_config(notebook_instance_lfc_name)
+            is not None
+        )
```

Comment:
Maybe we should make a delete notebook (using the sdk) to make sure we don't have any dangling resources after?

Reply:
That is done here: https://github.com/aws-controllers-k8s/sagemaker-controller/blob/ed370d2b6aad386c657f5f65ef3b06868be366ef/test/e2e/tests/test_retain_policy.py#L51